### PR TITLE
Add default settings to Build Monitor Views

### PR DESCRIPTION
**Pull request by timbrown5**
- Added settings items to config so default settings can be tracked
  in the plugin, not just the cookie.
- Updated configure-entries.jelly to allow setting of the settings.
- Updated main.settings.jelly to pass default settings from plugin to
  settings.js. Also added a new button to reset settings in cookie to
  default.
- Updated settings.js to use the default settings provided by
  main.settings.jelly. Also added a function to reset settings cookie
  to the default value.

I opened a new pull request as #419 was closed when I deleted the remote branch.

### CODE REVIEW

File: build-monitor-plugin/src/main/java/com/smartcodeltd/jenkinsci/plugins/buildmonitor/Config.java [hunk 1]
```diff
@@ -32,23 +37,63 @@ public static Config defaultConfig() {
...
+
+    public void setMaxColumns(int maxColumns) {
+        this.maxColumns = maxColumns;
+    }
+
+    public Double getTextScale() {
+        return getOrElse(textScale, 1.0);
+    }
+
+    public void setTextScale(Double scale) {
```
Comment (timja) [inline build-monitor-plugin/src/main/java/com/smartcodeltd/jenkinsci/plugins/buildmonitor/Config.java]: use the primitive?
```suggestion
    public void setTextScale(double scale) {
```

**Comment by timja** [comment]
Tests are failing, possibly legitimate failures?

**Comment by timbrown5** [comment]
> Tests are failing, possibly legitimate failures?
Yes, looks like when a new Build Monitor is created the default settings used are the ones defined in `Config.java`. The number of columns is defaulting to 0, which hides all the builds from the dashboard. I think allowing users to set 0 columns is also a bug, as the dashboard becomes useless.

**Comment by timbrown5** [comment]
Looks like the job passed but 4 tests failed to find objects using xpath.

**Comment by timbrown5** [comment]
> Looks like the job passed but 4 tests failed to find objects using xpath.

Looks like there are similar fialurs in master builds.
https://travis-ci.org/jan-molak/jenkins-build-monitor-plugin/jobs/596586973
https://travis-ci.org/jan-molak/jenkins-build-monitor-plugin/jobs/596538393

Is it that there are some flakey tests which get rerun?

**Comment by timbrown5** [comment]
@timja Thanks for the help.
Unless I hear otherwise I will assume this just needs merging.

**Comment by MarkTickner** [comment]
Is there any progress with this? This is a feature I'm looking forward to

**Comment by timja** [comment]
@jan-molak

**Comment by timbrown5** [comment]
Any update on when this might be merged?

**Comment by timbrown5** [comment]
@jan-molak Do you know when this might get merged?

**Comment by timbrown5** [comment]
Any update of when this will get merged?

**Pull request closed without merging**